### PR TITLE
CN-4875 wire hodl mode flow

### PR DESCRIPTION
**Pull request by ShamanShaulic**
## Checklist (Mandatory):

- [x] I have named the PR correctly (Check PR1)
- [x] Check the commit’s or even all commits message styles matches our requested structure. (issue number, etc.) (Check CR10)
- [x] I have tested new code locally (Check CR3)
- [x] I have done a smoke test of the larger feature locally (Check CR3)
- [x] I have added at least two reviewers (Check PR5)
- [x] I have assigned the PR to myself (Check PR4)
- [x] I have merged develop (parent) branch into my feature branch and resolved all conflicts (Check PR7)

## Checklist (Optional):

- [ ] I have added necessary documentation, including instructions how to test (Check CR8)
- [ ] I have added tests that prove my fix/implementation is effective or that my feature works
- [ ] I have written the code according to the adopted style guide? (Check CR7)

## Additional notes

> What does this implement/fix? Explain your changes. (optional)
> Are there any scripts that need running (eg. yarn install)?
> Are there any edge cases that need checking?
> Are there any specific test instructions?

[MVPWorkshop Pull Request & Code Review Guidelines](https://docs.google.com/document/d/1TcuU1JBhlGZLtpy0osSNinXQ2piP_yrDuRcXdqaOS_U/edit#heading=h.5b6kpdr4btq8)

### CODE REVIEW

File: app/components/atoms/HodlBanner/HodlBanner.js [hunk 1]
```diff
@@ -9,15 +10,32 @@ class HodlBanner extends Component {
...
+    if (!status.isActive) return null;
+
+    const now = moment.utc();
+    const deactivatedAt = moment.utc(status.deactivated_at);
+    const diff = deactivatedAt.diff(now);
+    const hours = Math.abs(moment.duration(diff).hours());
+    const minutes = Math.abs(moment.duration(diff).minutes());
+
+    const isDisabled =
+      status.state !== "Activated" || activeScreen === "VerifyProfile";
```
Comment (CrnogoracIvan) [inline app/components/atoms/HodlBanner/HodlBanner.js]: Move to constants.

File: app/components/atoms/HodlBanner/HodlBanner.js [hunk 2]
```diff
@@ -9,15 +10,32 @@ class HodlBanner extends Component {
...
+
     return (
-      <View style={style.container}>
-        {status ? (
+      <TouchableOpacity
+        style={style.container}
+        disabled={isDisabled}
+        onPress={() => navigateTo("HodlEmptyScreen")}
+      >
+        {status.state === "Activated" ? (
```
Comment (CrnogoracIvan) [inline app/components/atoms/HodlBanner/HodlBanner.js]: Constants?

File: app/components/screens/CheckYourEmail/CheckYourEmail.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import React, { Component } from "react";
```
Comment (CrnogoracIvan) [inline app/components/screens/CheckYourEmail/CheckYourEmail.js]: Try to add to Empty State screen

File: app/components/screens/HodlEmptyScreen/HodlEmptyScreen.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import React, { Component } from "react";
```
Comment (CrnogoracIvan) [inline app/components/screens/HodlEmptyScreen/HodlEmptyScreen.js]: Move to Empty state. Use setState if it is needed.

File: app/redux/hodl/hodlActions.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    const { formData } = getState().forms;
+
+    try {
+      const verification = {
+        pin: formData.pin,
+        twoFactorCode: formData.code,
+      };
+      await hodlService.activateHodlMode(verification);
+
+      dispatch(navigateTo("CheckYourEmail"));
```
Comment (CrnogoracIvan) [inline app/redux/hodl/hodlActions.js]: Check this, how to move to empty state.

File: app/redux/hodl/hodlActions.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    try {
+      // const hodlCode = formData.hodlCode;
+      const verification = {
+        pin: formData.pin,
+        twoFactorCode: formData.code,
+        code: formData.hodlCode,
+      };
+
+      await hodlService.deactivateHodlMode(verification);
+      dispatch(navigateTo("CheckYourEmail"));
```
Comment (CrnogoracIvan) [inline app/redux/hodl/hodlActions.js]: Check this, how to move to empty state.

File: app/components/screens/HODLViewCode/HODLViewCode.js [hunk 7]
```diff
@@ -5,36 +5,78 @@ import { connect } from "react-redux";
...
   });
 
+  componentWillUnmount() {
+    const { actions } = this.props;
+    actions.clearForm();
+  }
+
+  checkEmail = async () => {
+    const { actions } = this.props;
+    const response = await actions.activateHodlMode();
```
Comment (CrnogoracIvan) [inline app/components/screens/HODLViewCode/HODLViewCode.js]: Check with backend to return something usefully.

File: app/components/screens/HodlDeactivationCode/HodlDeactivationCode.js [hunk 8]
```diff
@@ -10,31 +10,70 @@ import RegularLayout from "../../layouts/RegularLayout/RegularLayout";
...
   });
 
+  componentWillUnmount() {
+    const { actions } = this.props;
+    actions.clearForm();
+  }
+
+  checkEmail = async () => {
+    const { actions } = this.props;
+    const response = await actions.deactivateHodlMode();
```
Comment (CrnogoracIvan) [inline app/components/screens/HodlDeactivationCode/HodlDeactivationCode.js]: Same as activateHodlMode
